### PR TITLE
Add Forms

### PR DESCRIPTION
**Pull request by okoyecharles**
## Description ✏️
In this project, I added forms to my Blog app.

## Changes Made ♻️
- [x] Create routes for likes and comments
- [x] Modify controllers for Posts
- [x] Create controllers for Comments and Likes
- [x] Render view for `posts/new`
- [x] Render view for `comments/new` & add like functionality
- [x] Resolve linter errors

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,11 +1,44 @@
...
-    @post_idx = @user.posts.find_index(@post) + 1
+    @post_idx = params[:id]
+  end
+
+  def new
+    @user = User.find(params[:user_id])
+    @post = @user.posts.new
+
+    render :new, locals: { post: @post }
+  end
```
Comment (mikethreels) [inline app/controllers/posts_controller.rb]: - [ ] for the new method in the Postcontroller you only need to add `@post = Post.new`

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,11 +1,44 @@
...
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = @user.posts.new(post_params)
```
Comment (mikethreels) [inline app/controllers/posts_controller.rb]: - [ ] this is what you created the `current_user` method for in the application_controller, you can write it as follows: `@post = @current_user.posts.new(post_params)`

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+class CommentsController < ApplicationController
+  def create
+    user = User.find(params[:user_id])
+    post = user.posts[params[:post_id].to_i - 1]
+    comment = Comment.new(comment_params)
+    comment.author = current_user
+    comment.post = post
```
Comment (mikethreels) [inline app/controllers/comments_controller.rb]: - [ ] for this part you created the `current_user` method, you can write it something like this:
```
    @comment = @current_user.comments.new(params)
    @comment.post_id = params[:post_id]
```

File: app/controllers/likes_controller.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+class LikesController < ApplicationController
+  def create
+    user = User.find(params[:user_id])
+    post = user.posts[params[:id].to_i - 1]
+    like = Like.new
+    like.author = current_user
+    like.post = post
```
Comment (mikethreels) [inline app/controllers/likes_controller.rb]: - [ ] in this method you can also use the current_user method created in the application controller.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Charles, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @OkoyeCharles, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You did a great job making the changes requested by the previous reviewer 👏👏

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**